### PR TITLE
Make it possible to provide per-provider topic

### PR DESCRIPTION
**Pull request by pnuu**
With this PR it is possible to provide a topic specific to each provider. If `topic` (for client) or `client_topic` (for mirror) is defined, it will be subscribed on every provider.

### CODE REVIEW

File: trollmoves/client.py [hunk 1]
```diff
@@ -407,6 +407,10 @@ def reload_config(filename, chains, callback=request_push, pub_instance=None):
             if val.get("heartbeat", False):
                 topics.append(SERVER_HEARTBEAT_TOPIC)
             for provider in chains[key]["providers"]:
+                if '/' in provider.split(':')[-1]:
```
Comment (mraspaud) [inline trollmoves/client.py]: shouldn't this be:
```suggestion
                if '/' in provider.split(':')[1]:
```

File: trollmoves/client.py [hunk 2]
```diff
@@ -407,6 +407,10 @@ def reload_config(filename, chains, callback=request_push, pub_instance=None):
...
+                    parts = urlparse(provider)
+                    provider = parts.scheme + '://' + parts.netloc
```
Comment (mraspaud) [inline trollmoves/client.py]: use `urlunparse` for this.

File: examples/move_it_mirror.ini [hunk 3]
```diff
@@ -62,3 +62,24 @@ topic = /1b/hrit-segment/0deg
...
 # delete = True
+
+[eumetcast-hrit-0deg_scp_per_server_topic]
+# client part, the incoming data, providers with topics
+providers = satmottag2:9011/topic/1 satmottag:9011/topic/2
+login = user
+# Destination where the data are saved locally
+destination = scp:///tmp/bar/
+# Topics listened for advertised incoming data from all providers. Optional.
+client_topic = /1b/hrit-segment/0deg
```
Comment (mraspaud) [inline examples/move_it_mirror.ini]: I'm not clear on the difference between this and what is specified in `providers`

File: bin/move_it_mirror.py [hunk 4]
```diff
@@ -103,8 +105,16 @@ class Listeners(object):
...
+        if client_topic is None:
+            topics = []
+        else:
+            topics = [client_topic]
         for provider in providers.split():
-            self.listeners.append(Listener('tcp://' + provider, [client_topic],
+            if '/' in provider:
+                parts = provider.split('/', 1)
+                provider = parts[0]
+                topics.append('/' + parts[1])
```
Comment (mraspaud) [inline bin/move_it_mirror.py]: So one provider can have multiple topics ?

**Comment by mraspaud** [comment]
Many questions here. I guess we need to write some documentation about this. Could you write a paragraph or two about operating a mirror with multiple topics ?

File: trollmoves/client.py (hunk 1 above)
Comment (pnuu) [reply trollmoves/client.py]: Here the `provider` has `tcp://` prefixed, so the last part of the provider string needs to be checked for the topic.

File: trollmoves/client.py (hunk 2 above)
Comment (pnuu) [reply trollmoves/client.py]: Using `urlunparse` would be unnecessarily complicated as the attributes can't be modified, and the topic (`parts.path`) needs to be removed from the provider.

File: examples/move_it_mirror.ini (hunk 3 above)
Comment (pnuu) [reply examples/move_it_mirror.ini]: `client_topic` can be considered as a global default topic to subscribe to. So some of the providers may have a specific topic, and for the others the global one would still be available.

File: bin/move_it_mirror.py (hunk 4 above)
Comment (pnuu) [reply bin/move_it_mirror.py]: The underlying `NSSubscriber` used by the client uses list of topics, so thought that would be the way to go. Do you think the `provider` topic, if specified, should overwrite the global topic?

**Comment by pnuu** [comment]
The mirror is still a bit of a mystery to me, but what I was thinking was:

- User has own servers, call them `own_a` and `own_b` serving the same data and acting as each others' backups, both servers using the same topic
- The `mirror` is used as a gateway to other external servers that can't be connected directly, and they might have different topics to what the local servers use
- The user wants to have the data from `own_a` and `own_b` directly without the data first being transferred to `mirror`

In this case the user can configure their client to listen to the topic used by local servers and the one published by the mirror, and wouldn't get duplicate data and would get the data with minimum delay as long as the local servers receive the data.

File: trollmoves/client.py (hunk 1 above)
Comment (mraspaud) [reply trollmoves/client.py]: Then you need to make sure that there is a scheme in the provider url
Maybe using urlsparse could help making sure we look at the right part ?

File: trollmoves/client.py (hunk 2 above)
Comment (mraspaud) [reply trollmoves/client.py]: I insist :) I think it'll make the code more robust.

File: bin/move_it_mirror.py (hunk 4 above)
Comment (mraspaud) [reply bin/move_it_mirror.py]: yes, otherwise it might lead to unwanted topics being monitored

**Comment by pnuu** [comment]
Ok, I now use more `urlparse`/`urlunparse` to format the provider URI.

**Pull request merged**